### ISSUE TITLE
Code coverage doesn't ever increment a line over 0

### ISSUE THREAD

**Issue by jcrist** [JuliaLang/julia#9891]
As discussed on [julia-users](https://groups.google.com/forum/#!topic/julia-users/pRaDS_4FB-4). I discovered this while trying to get coverage setup on a package I'm working on, but am able to reproduce it in this small test case:

With this code in `test.jl`:

```
function foo(x)
    return x + 1
end

x = 0
for i=1:100
    x = foo(x)
end
print(x)
```

I run `julia --code-coverage --inline=no test.jl`, which creates `test.jl.cov` as follows:

```
        - function foo(x)
        0     return x + 1
        - end
        - 
        - x = 0
        - for i=1:100
        -     x = foo(x)
        - end
        - print(x)
        - 
```

Clearly that coverage report isn't correct, as line 2 should have been hit 100 times, but it only reports 0. On a larger scale, with my test suite which runs some functions many, many times, I was never able to get a single line above 0. Most lines that I know are run result in `0`, while the rest remain at `-`.

Version info:
```
julia> versioninfo()
Julia Version 0.4.0-dev+2688
Commit eea31ae* (2015-01-14 07:25 UTC)
Platform Info:
  System: Linux (x86_64-unknown-linux-gnu)
  CPU: Intel(R) Core(TM)2 Duo CPU     T6500  @ 2.10GHz
  WORD_SIZE: 64
  BLAS: libblas
  LAPACK: liblapack
  LIBM: libm
  LLVM: libLLVM-3.5.0
```

**Comment by timholy** [comment]
I wouldn't be shocked if this is an LLVM-3.5 thing. Possibly-related: #7910.

**Comment by simonster** [comment]
Presumably this was a dup of #7779, which @vtjnash fixed in 722b35a25e003d46cb5a0dc001ce6f0f933f9ef4
